sprint3/ProductApi.js

```diff
@@ -0,0 +1,373 @@
+    const id = Number(req.params.id) ;
+
+    if (!id){
+        const err = new Error("invalid parameter")
+        err.status = 400;
+        return next(err);
+    }
+
+    if (!name|| !description ||!price || !tags){
+        const err = new Error("invalid body data");
+        err.status = 400;
+        return next(err);
+    }
+
+    try{
+        const product= await prisma.product.update({
+            where: {id},
+            data: {
+                name,
+                description,
+                price,
+                tags
+            }
+        });
+        console.log("patch success");
+        return res.status(200).send(product);
+        
+    }catch(error){
+        console.log('patch product failed because of server error');
+        const err = new Error("Server Error");
+        err.status = 500;
+        return next(err);
+    }
+});
+
+//게시글 삭제하기 
+ProductRouter.delete('/detail/:id', async (req,res,next) =>{
+    const id = Number(req.params.id) ;
+    if (!id){
+        const err = new Error("invalid parameter")
+        err.status = 400;
+        return next(err);
+    }
+
+    const product = await prisma.Product.findUnique({
+            where:{id}
+    });
+    if (!product){
+        const err = new Error("No content")
+        err.status = 404;
+        return next(err);
+    }
+
+    try{
+        await prisma.Product.delete({
+            where:{id}
+        });
+        console.log("deleting success");
+        return res.status(204).send("deleting successed");
+        
+    }catch(error){
+        console.log('deleting product failed because of server error');
+        const err = new Error("Server Error");
+        err.status = 500;
+        return next(err);
+    }
+});
+
+
+
+
+
+//------------------------------댓글-----------------
+
+
+
+//모든 댓글 보기
+ProductRouter.get('/comments', async (req,res,next) =>{
+    try{
+        let {take = '10',skip= '1',commentId = '1'} = req.query;
+        take = parseInt(take);
+        skip = parseInt(skip);
+        commentId = parseInt(commentId);
+
+        const comments= await prisma.ProductComment.findMany({
+            take,
+            skip,
+            cursor: {
+                id: commentId
+            },
+            orderBy:{
+                id: 'asc'
+            }
+        });
+        if (!comments){
+            return res.status(300).send("There isn't comment. Write the first comment!");
+        }
+        
+        return res.status(200).send(comments);
+    }catch(error){
+        console.error(error);
+        const err = new Error("Server Error");
+        err.status = 500;
+        return next(err);
+    }
+});    
+
+    
+
+
+
+
+//게시글 상세페이지에서 댓글 달기
+ProductRouter.post('/detail/:id', async (req,res,next) =>{
+    let id;
+    id = Number(req.params.id) ;
+
+    if (!id){
+        const err = new Error("invalid parameter")
+        err.status = 400;
+        return next(err);
+    }
+
+    const product = await prisma.Product.findUnique({
```

Comment:
아래처럼 스키마 부분을 소문자로 작성하지 않아도 잘 동작하는지 확인해주세요.

```suggestion
    const product = await prisma.product.findUnique({
```